context/app/static/js/components/profile/Summary/hooks.ts

```diff
@@ -0,0 +1,70 @@
+import { readCookie } from 'js/helpers/functions';
+import { fetcher } from 'js/helpers/swr';
+import { SWRError } from 'js/helpers/swr/errors';
+import useSWRImmutable from 'swr/immutable';
+
+// The globus auth helper from hubmap_commons omits the group descriptions, so we need to fetch the full list of groups from the repo.
+const globusGroupsURL = '/api/globus-groups.json';
+
+interface GlobusGroupInfo {
+  description: string;
+  data_provider: boolean;
+  has_subgroups: boolean;
+  identity_set_properties: null; // This is always null in the data.
+  uuid: string;
+  group_type: 'regular' | 'data-read' | 'protected-data' | 'data-admin';
+  name: string;
+  displayname: string;
+  shortname: string;
+  generateuuid: boolean;
+  tmc_prefix: string;
+}
+
+export function useGlobusGroups(shouldRequest = true) {
+  const { data, error } = useSWRImmutable<GlobusGroupInfo[], SWRError>(
+    shouldRequest ? globusGroupsURL : null,
+    (url: string) => fetcher<GlobusGroupInfo[]>({ url }),
+  );
+
+  return {
+    data,
+    error,
+    isLoading: !data && !error,
+  };
+}
+
+export function useCurrentUserGlobusGroups() {
+  const groupsCookie = readCookie('user_groups');
```

Comment:
We could consider adding `user_groups` passed from flask to the context.

Reply:
We could; I figured we should have this be consistent with the existing approach for tracking where we use the cookie. Are there any drawbacks you can think of for this approach?